### ISSUE TITLE
Upgrade to Listr 2

### ISSUE THREAD

**Issue by from-nibly** [tsedio/tsed-cli#153]
It looks like the listr project has been abandoned. any chance this could use listr2. It looks like it might be a drop in replacement.

https://github.com/cenk1cenk2/listr2
https://listr2.kilic.dev/

**Comment by Romakita** [comment]
You can make a PR ;)

**Comment by from-nibly** [comment]
OK so I tried upgrading, and it was not a 100% drop in replacement. I changed a couple things, but I'm stuck on two errors while running the tests. 


First, it seems that the setRenderer function isn't present on whatever the result of this function call is.

[src/services/ProjectPackageJson.spec.ts](https://github.com/tsedio/tsed-cli/blob/master/packages/cli-core/src/services/ProjectPackageJson.spec.ts#L65)
```ts
      const list = projectPackageJson.install({packageManager: PackageManager.NPM}); 
```

Second, It's also throwing an uncaught exception error. 
Saying ` TypeError: Converting circular structure to JSON`

The stack trace on this is not really helpful I tried doing some console.log statements around all of the various `JSON.stringify` calls in the project but I think something in jest is silencing the console.log statements, or it's not happening inside the cli repo.

I'm having issues using my normal debugging tools (console.log) . I'll see if I can spend some more time on this, but I'm not sure I'll be able to.

I'm mostly interested in the Listr2 support for exactly this issue, I'm having trouble developing CLI commands with tsed-cli because listr v1 doesn't really handle printing to the console very well. and Listr v2 seems to have support for using your own logger.
